### DISCUSSION TITLE
Resource menu items for subsets

### DISCUSSION THREAD

**Discussion by LemonPie22** [filamentphp/filament#6969] [Help] [answered]
I have a `users` resource which shows customers, staff and users with other roles.

Is there a way - without creating (and having to maintain) separate resources - to offer specific menu items for customers and staff?

From this:
<img width="316" alt="Screenshot 2023-07-09 at 13 41 27" src="https://github.com/filamentphp/filament/assets/107488019/5dc4608f-965c-4f45-bf1c-9270315f359d">

To this:
<img width="207" alt="Screenshot 2023-07-09 at 13 49 28" src="https://github.com/filamentphp/filament/assets/107488019/516c7225-fe4a-4e8c-9d84-280f8ec095a2">

I'm thinking that it may be achievable using pre-set filters but I couldn't figure out how.

**Comment by danharrin** [comment]
I don't know what you exactly want to do, please elaborate

**Comment by LemonPie22** [comment]
Okay I'll try and clarify.

I have a standard Filament resource that allows me to list users:
<img width="895" alt="Screenshot 2023-07-10 at 14 40 06" src="https://github.com/filamentphp/filament/assets/107488019/a37f114b-40d9-4289-82c2-15313bd10b69">

It looks like this:
<img width="443" alt="Screenshot 2023-07-10 at 14 42 36" src="https://github.com/filamentphp/filament/assets/107488019/fbcf45bc-31cd-4e66-ace8-6fb8b326b5f7">

All pretty standard stuff.

I also have a number of roles:
<img width="607" alt="Screenshot 2023-07-10 at 14 45 29" src="https://github.com/filamentphp/filament/assets/107488019/fd899be3-75e4-4378-a5f9-8949c7e889b4">

And each user can have multiple roles:
<img width="660" alt="Screenshot 2023-07-10 at 14 47 00" src="https://github.com/filamentphp/filament/assets/107488019/a08c97f1-0c16-4008-af7d-256c17cd9e72">

This last is achieved in the user resource like this:
```
Forms\Components\Select::make('roles')
    ->multiple()
    ->relationship(
        'roles',
        'name',
        // Exclude super-admin role from list for non super-admins
        function (Builder $query) {
            if (Auth::user()->hasRole('super-admin')) {
                return $query
                ->orderBy('id');
            }
            return $query
            ->where('name', '!=', 'super-admin')
            ->orderBy('id');
        }
    )
    ->preload(),
```

All this works fine but I want to have a menu item called "Customers" that only lists users with the role `customer`. And I want another menu item called "Staff" that lists users that don't have `customer` as a role.

I realise I could triplicate my users resource, but then I'd have 3 almost identical resources to maintain.

I know I could just filter my users by role to see customers, etc. but that's several more steps than I want.

I want a menu item called "Customers" that, when clicked, opens the users resource but filtered to only show customers. And another called "Staff" that only shows non-customers.

I suspect this is probably really simple and I'm missing something obvious.

**Reply by danharrin** [reply, marked as answer]
You can create the extra CustomerResource and StaffResource, extend the main UserResource, set `$slug` on each, and then override `getEloquentQuery()` on each to scope the query. Then you only need to maintain 1 `form()` and `table()`

**Reply by LemonPie22** [reply]
Thanks @danharrin, that works perfectly. For anyone with the same issue, here's the code I used in my CustomerResource:
```
<?php

namespace App\Filament\Resources;

use App\Filament\Resources\CustomerResource\Pages;
use App\Filament\Resources\CustomerResource\RelationManagers;
use App\Models\User;
use Illuminate\Database\Eloquent\Builder;
use Illuminate\Database\Eloquent\SoftDeletingScope;

class CustomerResource extends UserResource
{
    protected static ?string $modelLabel = 'customer';
    protected static ?string $slug = 'customer';

    protected static ?string $navigationIcon = 'heroicon-o-collection';

    public static function getPages(): array
    {
        return [
            'index' => Pages\ListCustomers::route('/'),
            'create' => Pages\CreateCustomer::route('/create'),
            'edit' => Pages\EditCustomer::route('/{record}/edit'),
        ];
    }

    public static function getEloquentQuery(): Builder
    {
        return parent::getEloquentQuery()->with('roles')
        ->whereHas('roles', function ($q) {
            $q->whereIn('name', ['customer']);
        })
        ->withoutGlobalScopes(
            [
                SoftDeletingScope::class,
            ]
        );
    }
}
```
